package-build.el

```diff
@@ -1080,9 +1094,10 @@ and a cl struct in Emacs HEAD.  This wrapper normalises the results."
 
     (package-build--message "\n;;; %s\n" file-name)
     (let* ((version (package-version-join
-                     (version-to-list
+                     (package-build--version-to-list
```

Comment:
Yes exactly. It took me a second to figure out why we had the two functions to begin with and I'm still not sure I know but they're getting consolidated anyhow. I'm not sure how it'll work though with something like hg where you kinda need two regex. One to pull the tag and one to pull the version from the tag. However I think we can hardcore the "tag" regex and then use version-regexp to parse the version from the tag. ?